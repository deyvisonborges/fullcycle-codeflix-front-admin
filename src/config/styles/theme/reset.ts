import { createGlobalStyle } from 'styled-components'

const ResetStyles = createGlobalStyle`
  * {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
  }

  html, body {
    font-family: 'Poppins', sans-serif;
    min-height: 100vh;
  }

  html, body, #root {
    height: 100%;
  }
`
export default ResetStyles
