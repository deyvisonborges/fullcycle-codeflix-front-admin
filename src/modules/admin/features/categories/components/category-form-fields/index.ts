export * from './category-form-fields'
