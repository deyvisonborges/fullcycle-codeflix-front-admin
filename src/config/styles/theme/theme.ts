import { DefaultTheme } from 'styled-components'
import { ThemeMode } from './theme.types'

export const lightTheme: DefaultTheme = {
  informative: { danger: '#d72c0d' },
  background: {
    default: 'yellow'
  },
  primary: {
    main: '#008060'
  },
  secondary: {
    main: '#111111'
  },
  text: {
    primary: '#222222'
  }
}

export const darkTheme: DefaultTheme = {
  informative: { danger: '#d72c0d' },
  background: {
    default: '#222222'
  },
  primary: {
    main: '#008060'
  },
  secondary: {
    main: '#FFFFFF'
  },
  text: {
    primary: '#FFFFFF'
  }
}

export const theme: Record<ThemeMode, DefaultTheme> = {
  light: lightTheme,
  dark: darkTheme
}
