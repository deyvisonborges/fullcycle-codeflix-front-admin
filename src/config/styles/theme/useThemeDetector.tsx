import { useEffect, useState } from 'react'
import { ThemeMode } from './theme.types'

// https://theme-toggle.rdsx.dev/?ref=dailydev
// transicoes de pagina

export function useThemeSystemDetector() {
  const isClient = typeof window !== 'undefined'

  const getCurrentTheme = () =>
    isClient && window.matchMedia('(prefers-color-scheme: dark)').matches
      ? 'dark'
      : 'light'

  const [theme, settheme] = useState<ThemeMode>(getCurrentTheme())

  useEffect(() => {
    if (!isClient) return

    const darkThemeMq = window.matchMedia('(prefers-color-scheme: dark)')

    const themeChangeHandler = (event: MediaQueryListEvent) => {
      settheme(event.matches ? 'dark' : 'light')
    }

    darkThemeMq.addEventListener('change', themeChangeHandler)
    return () => darkThemeMq.removeEventListener('change', themeChangeHandler)
  }, [isClient])

  return theme
}
