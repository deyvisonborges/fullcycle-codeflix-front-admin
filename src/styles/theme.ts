import { DefaultTheme } from 'styled-components'
import { ThemeMode } from './theme.provider'

export const lightTheme: DefaultTheme = {
  background: {
    default: 'yellow'
  },
  primary: {
    main: '#111111'
  },
  secondary: {
    main: '#111111'
  },
  text: {
    primary: '#222222'
  }
}

export const darkTheme: DefaultTheme = {
  background: {
    default: '#222222'
  },
  primary: {
    main: '#E50914'
  },
  secondary: {
    main: '#FFFFFF'
  },
  text: {
    primary: '#FFFFFF'
  }
}

export const theme: Record<ThemeMode, DefaultTheme> = {
  light: lightTheme,
  dark: darkTheme
}
