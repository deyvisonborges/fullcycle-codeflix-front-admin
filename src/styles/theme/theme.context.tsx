import { DefaultTheme } from 'styled-components'
import { createContext } from 'react'
import { ThemeMode } from './theme.types'

export type ThemeContextDataProps = {
  theme: DefaultTheme | undefined
  changeTheme: (mode: ThemeMode) => void
}

export const ThemeContextData = createContext({} as ThemeContextDataProps)
