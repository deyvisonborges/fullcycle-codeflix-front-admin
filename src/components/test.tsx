import * as S from './test.styles'
import { toastActions, useAppDispatch, useAppSelector } from '@/store'

export function Test() {
  const dispatch = useAppDispatch()
  const { isVisible, message } = useAppSelector((state) => state.toast)
  return (
    <div>
      <p>Habilitado: {isVisible}</p>
      {isVisible && <p>Message: {message}</p>}
      <button
        onClick={() =>
          isVisible
            ? dispatch(toastActions.hideToast())
            : dispatch(toastActions.showToast(`Messagem para o toast`))
        }
      >
        {isVisible ? 'Desabilitar' : 'Habilitar'}
      </button>
      <S.Button as="a" href="/dashboard">
        Button
      </S.Button>
      <a href="/not-fount">Not found</a>
    </div>
  )
}
