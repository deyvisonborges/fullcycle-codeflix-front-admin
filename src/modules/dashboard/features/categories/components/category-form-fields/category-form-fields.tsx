import * as S from './category-form-fields.styles'
import { Input } from '@/components/input'
import { CategoryUIModel } from '../../category.ui-model'

type CategoryFormFieldsDataProps = Omit<
  CategoryUIModel,
  'id ' | 'deletedAt' | 'createdAt' | 'updatedAt'
>

type CategoryFormFieldsProps = {
  data: CategoryFormFieldsDataProps
  handleChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  handleToggle: (e: React.ChangeEvent<HTMLInputElement>) => void
  isDisabled?: boolean
}

export function CategoryFormFields({
  isDisabled,
  data,
  handleChange,
  handleToggle
}: CategoryFormFieldsProps) {
  return (
    <>
      <S.InputGroup>
        <Input
          id="name"
          type="text"
          name="name"
          placeholder="Nome da categoria"
          value={data.name || ''}
          label="Nome"
          disabled={isDisabled}
          onChange={handleChange}
        />
        <Input
          id="description"
          type="text"
          name="description"
          placeholder="Descrição"
          value={data.description || ''}
          label="Descrição"
          disabled={isDisabled}
          onChange={handleChange}
        />
      </S.InputGroup>

      <Input
        id="is_active"
        type="checkbox"
        name="is_active"
        defaultChecked={data.isActive ?? false}
        label="Ativo?"
        disabled={isDisabled}
        onChange={handleToggle}
      />
    </>
  )
}
