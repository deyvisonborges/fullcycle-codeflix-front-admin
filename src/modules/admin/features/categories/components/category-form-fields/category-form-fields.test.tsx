import { ThemeProvider } from '@/config/styles/theme/theme.provider'
import { CategoryFormFields } from './category-form-fields'
import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { mockMatchMedia } from '@/utils/test/mock-matchmedia'

const mockProps = {
  isDisabled: true,
  handleChange: jest.fn(),
  handleToggle: jest.fn()
}

// é um hook de configuração que executa um bloco de código antes de todos os testes dentro de um arquivo de teste serem executados.
beforeAll(() => mockMatchMedia)

describe('CategoryFormFields', () => {
  it('should render category form fields correctly', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <CategoryFormFields
          data={{
            name: 'mock name',
            description: 'some mock description',
            isActive: true
          }}
          {...mockProps}
        />
      </ThemeProvider>,
      { wrapper: BrowserRouter }
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render category from with disabled state', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <CategoryFormFields
          data={{
            name: 'mock name',
            description: 'some mock description',
            isActive: true
          }}
          {...mockProps}
          isDisabled
        />
      </ThemeProvider>,
      { wrapper: BrowserRouter }
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
