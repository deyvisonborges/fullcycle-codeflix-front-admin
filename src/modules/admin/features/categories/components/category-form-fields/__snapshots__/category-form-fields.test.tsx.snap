// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryFormFields should render category form fields correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb blRQuv"
  >
    <div
      class="sc-gtLWhw kFIqRo"
    >
      <label
        class="sc-fAUdSK kTGwm"
        for="name"
      >
        Nome
      </label>
      <input
        class="sc-egkSDF emYusT"
        disabled=""
        id="name"
        name="name"
        placeholder="Nome da categoria"
        type="text"
        value="mock name"
      />
    </div>
    <div
      class="sc-gtLWhw kFIqRo"
    >
      <label
        class="sc-fAUdSK kTGwm"
        for="description"
      >
        Descrição
      </label>
      <input
        class="sc-egkSDF emYusT"
        disabled=""
        id="description"
        name="description"
        placeholder="Descrição"
        type="text"
        value="some mock description"
      />
    </div>
  </div>
  <div
    class="sc-gtLWhw kFIqRo"
  >
    <label
      class="sc-fAUdSK kTGwm"
      for="isActive"
    >
      Ativo?
    </label>
    <input
      checked=""
      class="sc-egkSDF emYusT"
      disabled=""
      id="isActive"
      name="isActive"
      type="checkbox"
    />
  </div>
</DocumentFragment>
`;

exports[`CategoryFormFields should render category from with disabled state 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb blRQuv"
  >
    <div
      class="sc-gtLWhw kFIqRo"
    >
      <label
        class="sc-fAUdSK kTGwm"
        for="name"
      >
        Nome
      </label>
      <input
        class="sc-egkSDF emYusT"
        disabled=""
        id="name"
        name="name"
        placeholder="Nome da categoria"
        type="text"
        value="mock name"
      />
    </div>
    <div
      class="sc-gtLWhw kFIqRo"
    >
      <label
        class="sc-fAUdSK kTGwm"
        for="description"
      >
        Descrição
      </label>
      <input
        class="sc-egkSDF emYusT"
        disabled=""
        id="description"
        name="description"
        placeholder="Descrição"
        type="text"
        value="some mock description"
      />
    </div>
  </div>
  <div
    class="sc-gtLWhw kFIqRo"
  >
    <label
      class="sc-fAUdSK kTGwm"
      for="isActive"
    >
      Ativo?
    </label>
    <input
      checked=""
      class="sc-egkSDF emYusT"
      disabled=""
      id="isActive"
      name="isActive"
      type="checkbox"
    />
  </div>
</DocumentFragment>
`;
